assets/src/js/main.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Main scripts, loaded on all pages.
+ *
+ * @package wp-mentions-links
+ */
+
+/**
+ * Post and User Autocompleter hooks.
+ *
+ * @see https://developer.wordpress.org/block-editor/components/autocomplete/
+ */
+
+const excludedBlocks = [];
+
+const postCompleter = {
+	name: 'posts-completer',
+	triggerPrefix: '#',
+	isDebounced: true,
+	options: async ( search ) => {
+		let queryString = '';
+		if ( search ) {
+			queryString = '?search=' + encodeURIComponent( search );
+		}
+
+		let returnArray = [];
+		if ( 'undefined' !== typeof wpMentionsLinks.supportedCPTs ) {
+			for ( let i in wpMentionsLinks.supportedCPTs ) {
+				returnArray = returnArray.concat( await wp.apiFetch( { path: '/wp/v2/' + wpMentionsLinks.supportedCPTs[i] + '' + queryString } ) );
+			}
+		} else {
+			returnArray = returnArray.concat( await wp.apiFetch( { path: '/wp/v2/posts' + queryString } ) );
+			returnArray = returnArray.concat( await wp.apiFetch( { path: '/wp/v2/pages' + queryString } ) );
+		}
+
+		return returnArray;
+	},
+	getOptionKeywords: ( post ) => post.title.rendered.split( /\s+/ ),
+	getOptionLabel: ( post ) => [
+		<span key={ post.id.toString() + '-icon' } className="dashicons dashicons-text-page" />,
```

Comment:
can we use https://codex.wordpress.org/Javascript_Reference/wp.template to render HTML?

Reply:
This HTML code is not going to increase in future. So I feel it's a very small snippet to be used as `wp.template`.

Reply:
Ok keep it as it is for now.